docs/docsite/rst/guide_modulehelper.rst

```diff
@@ -0,0 +1,520 @@
+..
+  Copyright (c) Ansible Project
+  GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+  SPDX-License-Identifier: GPL-3.0-or-later
+
+.. _ansible_collections.community.general.docsite.guide_modulehelper:
+
+Module Helper guide
+===================
+
+
+Introduction
+^^^^^^^^^^^^
+
+Writing a module for Ansible is largely described in existing documentation.
+However, a good part of that is boilerplate code that needs to be repeated every single time.
+That is where ``ModuleHelper`` comes to assistance: a lot of that boilerplate code is done.
+
+
+Quickstart
+""""""""""
+
+See the `example from Ansible documentation <https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_general.html#creating-a-module>`_
+written with ``ModuleHelper``.
+But bear in mind that it does not showcase all of MH's features:
+
+.. code-block:: python
+
+    from ansible_collections.community.general.plugins.module_utils.module_helper import ModuleHelper
+
+
+    class MyTest(ModuleHelper):
+        module = dict(
+            argument_spec=dict(
+                name=dict(type='str', required=True),
+                new=dict(type='bool', required=False, default=False),
+            ),
+            supports_check_mode=True,
+        )
+        use_old_vardict = False
+
+        def __run__(self):
+            self.vars.original_message = ''
+            self.vars.message = ''
+            if self.check_mode:
+                return
+            self.vars.original_message = self.vars.name
+            self.vars.message = 'goodbye'
+            self.changed = self.vars['new']
+            if self.vars.name == "fail me":
+                self.do_raise("You requested this to fail")
+
+
+    def main():
+        MyTest.execute()
+
+
+    if __name__ == '__main__':
+        main()
+
+
+Module Helper
+^^^^^^^^^^^^^
+
+Introduction
+""""""""""""
+
+``ModuleHelper`` is a wrapper around the standard ``AnsibleModule``, providing extra features and conveniences.
+The basic structure of a module using ``ModuleHelper`` is as shown in the `Quickstart`_ section above,
```

Comment:
Please add a proper label for that section (it need to have a `ansible_collections.community.general.docsite.` prefix) and use `:ref:` here to reference it.

Same for other references in this document.

Reply:
There's more of those in the doc, will review then.

Reply:
Done